### PR TITLE
Telemetry: Fix delayed init events

### PR DESCRIPTION
**Pull request by JReinhold**
Closes #

## What I did

We found that during init, telemetry events would not be sent until the very end of the flow, where everything would be sent at once. If the flow was canclled mid-way, no events would be sent.

This change ensures that events are sent when they happen.

This was originally a regression from https://github.com/storybookjs/storybook/pull/34485 which was released in 10.4.0-alpha.10, so it's only a small window of bad telemetry.

Now, we immediately resolve telemetry state, instead of relying on the different commands do to it eventually from their loaded presets.
There's a small change here, where previously we would use preset loading mechanism to look up `core.disableTelemetry` but now we just load the main config, because it's faster. So theoretically someone could have an external preset where they set telemetry setting and that wouldn't be supported anymore, now we only support it when it's directly set in main.ts (although it can be imported from elsewhere, etc.)

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

> [!CAUTION]
> This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!

-

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/publish.yml) or locally with `gh workflow run --repo storybookjs/storybook publish.yml --field pr=<PR_NUMBER>`_

## Summary by CodeRabbit

* **Refactor**
  * Telemetry preference is resolved earlier during startup and now uses a defined fallback behavior for unresolved cases, making initialization telemetry more consistent.
* **Tests**
  * Expanded telemetry initialization tests to cover CLI opt‑in, config‑derived settings, fallback behavior, and related edge cases.
* **Chores**
  * Test setup updated to ignore a specific Browserslist warning pattern.

### CODE REVIEW

**Review by valentinpalkovic — APPROVED** [review]

**Review by yannbf — APPROVED** [review]

File: code/core/src/core-server/withTelemetry.ts [hunk 1]
```diff
@@ -167,44 +166,37 @@ export async function sendTelemetryError(
...
-/**
- * Resolve telemetry state by loading presets from configDir to check core.disableTelemetry.
- * Used when run() completes without resolving telemetry state (e.g. CLI commands like
- * add/remove/doctor/upgrade/migrate that don't load presets themselves).
- */
-async function tryResolveTelemetryStateFromConfig(options: TelemetryOptions) {
-  const configDir = options.cliOptions.configDir || options.presetOptions?.configDir;
+async function resolveTelemetryState(options: TelemetryOptions) {
+  // 1. If telemetry is explicitly set via CLI options or env var, set and skip loading main config
+  if (options.cliOptions.disableTelemetry !== undefined) {
```
Comment (yannbf) [inline code/core/src/core-server/withTelemetry.ts:171]: does this account for the environment variable as well? should it? (in case resolving main.js fails in the next step)

**Review by yannbf — APPROVED** [review]

File: code/core/src/core-server/withTelemetry.ts (hunk 1 above)
Comment (JReinhold) [reply code/core/src/core-server/withTelemetry.ts:171]: Yes it does (the comment even say that 😆)
"CLI options" immediately fallback to env vars too, when they are collected at the very beginning
